### PR TITLE
Refactor/standardize most code that prints out team names

### PR DESCRIPTION
**Pull request by clifffrey**
This tries to standardize, within reason, how we print team names.

The user-visible changes here are that team names are logged to the console and to discord more consistently, are more often colored correctly with north/south colors, and whether "team" is used as a prefix before a team name is also more consistent.

Background for this change:
 1. There are "force names" which are "north" and "south"
 2. There are the bb_config.(north|south)_side_team_name defaults "Team North" and "Team South"
 3. There are the configured global.tm_custom_name[] overridden values. Not overridden by default, but in captains games, these generally are.

Before this change, there was a combination of different things above that were used when printing out a message, often with "team " prepended. Specifically, it was common to print out ("team " .. (tm_custom_name or force_name)) which would generally work ok. However, the inconsistency was strange. Also, it was unclear when choosing a custom team name if it should be "Team BuildMixers" or "The BuildMixers" or just "BuildMixers", and it would sometimes be printed out with "team " in front of it and sometimes not.

After this change, we try and relatively consistently use the new Functions.team_name API to print out a team name, and do not prepend it with "team " (because in the default case, it will be "Team North" or "Team South"). A notable exception to this is when logging to discord, we often print out ("team " .. force_name), because discord won't support the rich-text that is often used in custom team names.

Additionally, this adds team-name-coloring to the console output in a few more places (like when a player joins a team, or when a technology is researched).

For an added bonus, this change is net-negative in terms of lines of code.

A potential risk with this change is that it changes what is logged to discord. If there is code that is parsing the discord messages, it could be broken by this change.

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

**Comment by Ragnarok77-factorio** [comment]
Like the idea.

And yes, will need to check discord before testing it, as several bots are parsing the log. So it will be delayed quite a bit the testing of it.

**Comment by clifffrey** [comment]
ok, if there are things parsing the discord messages, how about I take a stab at making this change in a way that just doesn't change the discord messages at all (or as close to that as I can get). And we can look into making other changes later on that modify the discord messages if we want to?

**Comment by clifffrey** [comment]
Ok, I updated the change to essentially not change any Discord messages (except for one, which I think is very low risk, but I can make that one unchanged if people would be more comfortable), which I think substantially derisks this change.

File: maps/biter_battles_v2/feeding.lua [hunk 1]
```diff
@@ -67,25 +67,16 @@ local function get_enemy_team_of(team)
...
 	local formatted_food = table.concat({"[color=", food_values[food].color, "]", food_values[food].name, " juice[/color]", "[img=item/", food, "]"})
 	local formatted_amount = table.concat({"[font=heading-1][color=255,255,255]" .. flask_amount .. "[/color][/font]"})
 	
 	if flask_amount >= 20 then
-		local enemy = get_enemy_team_of(player.force.name)
-		game.print(table.concat({colored_player_name, " fed ", formatted_amount, " flasks of ", formatted_food, " to team ", team_strings[enemy], " biters!"}), {r = 0.9, g = 0.9, b = 0.9})
-		Server.to_discord_bold(table.concat({player.name, " fed ", flask_amount, " flasks of ", food_values[food].name, " to team ", enemy, " biters!"}))
+		game.print(table.concat({colored_player_name, " fed ", formatted_amount, " flasks of ", formatted_food, " to ", Functions.team_name_with_color(enemy), "'s biters!"}), {r = 0.9, g = 0.9, b = 0.9})
+		Server.to_discord_bold(table.concat({player.name, " fed ", flask_amount, " flasks of ", food_values[food].name, " to team ", enemy, "'s biters!"}))
```
Comment (developer-8) [inline maps/biter_battles_v2/feeding.lua:79]: this change of discord message may break send stats at https://bb.patrol.neutrino.re/current/

Comment (clifffrey) [reply maps/biter_battles_v2/feeding.lua]: Gah, yes, you are 100% right.

I am still getting used to VS code and I had not saved the changes that I had made. So I hadn't pushed the most recent version of the code.

I've fixed this (and a few other cases) now.

**Review by XVhc6A — APPROVED** [review]
great changes.  I cannot speak w/r/t the discord bot, but code appears sound.  Tested in single-player for a few minutes - didn't see any lua errors logged or anything wrong printed. 
image

**Review by developer-8 — APPROVED** [review]

**Review by Ragnarok77-factorio — CHANGES_REQUESTED** [review]
Conflicts in game-over.lua needs to be fixed first

**Comment by clifffrey** [comment]
fixed (I did a rebase rather than a merge, let me know if you'd prefer merges in the future)

**Pull request merged**